ClientInvocationFutureTest test_applyEither

- Fails on `Hazelcast-4..maintenance-ZingJDK8`
- Fails on [Build #183 (Jun 16, 2020 10:56:24 PM)](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4..maintenance-ZingJDK8/183/testReport/junit/com.hazelcast.client.impl.spi.impl/ClientInvocationFutureTest/test_applyEither/)
- Error
```
expected:<null> but was:<ClientMessage{connection=null, length=42, operation=null, isRetryable=false, correlationId=0, messageType=10201, isEvent=false, isfragmented=false}>
```
- Stacktrace
```
ava.lang.AssertionError: expected:<null> but was:<ClientMessage{connection=null, length=42, operation=null, isRetryable=false, correlationId=0, messageType=10201, isEvent=false, isfragmented=false}>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.client.impl.spi.impl.ClientInvocationFutureTest.test_applyEither(ClientInvocationFutureTest.java:295)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```
- Possible commits
```
CP Client message tasks should not deserialize responses (detail / githubweb)
[BACKPORT] Fixes interoperability issues with CompletableFuture (#17048) (detail / githubweb)
```

kindly check.

--- a/hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/ClientInvocationFutureTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/ClientInvocationFutureTest.java
@@ -18,11 +18,11 @@ package com.hazelcast.client.impl.spi.impl;
 
 import com.hazelcast.client.impl.protocol.ClientMessage;
 import com.hazelcast.client.impl.protocol.codec.MapGetCodec;
+import com.hazelcast.internal.serialization.Data;
 import com.hazelcast.internal.serialization.SerializationService;
 import com.hazelcast.internal.serialization.impl.DefaultSerializationServiceBuilder;
 import com.hazelcast.internal.util.RootCauseMatcher;
 import com.hazelcast.logging.ILogger;
-import com.hazelcast.internal.serialization.Data;
 import com.hazelcast.spi.impl.InternalCompletableFuture;
 import com.hazelcast.spi.impl.sequence.CallIdSequence;
 import com.hazelcast.test.HazelcastParallelClassRunner;
@@ -43,7 +43,6 @@ import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
 import static com.hazelcast.spi.impl.InternalCompletableFuture.newCompletedFuture;
-import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
 import static com.hazelcast.test.HazelcastTestSupport.ignore;
 import static com.hazelcast.test.HazelcastTestSupport.sleepSeconds;
 import static org.junit.Assert.assertEquals;
@@ -236,7 +235,7 @@ public class ClientInvocationFutureTest {
     public void test_thenAcceptBoth() throws Exception {
         CompletableFuture nextStage = invocationFuture.thenAcceptBoth(newCompletedFuture(null),
                 (t, u) -> ignore(null));
-        invocationFuture.complete(response);
+        invocationFuture.complete(null);
 
         assertEquals(null, nextStage.get(10, TimeUnit.SECONDS));
         verify(callIdSequence).forceNext();
@@ -278,29 +277,25 @@ public class ClientInvocationFutureTest {
     public void test_acceptEither() throws Exception {
         CompletableFuture nextStage = invocationFuture.acceptEither(newCompletedFuture(null),
                 t -> ignore(null));
-        invocationFuture.complete(response);
 
         assertEquals(null, nextStage.get(10, TimeUnit.SECONDS));
         verify(callIdSequence).forceNext();
-        verify(callIdSequence, times(2)).complete();
+        verify(callIdSequence, times(1)).complete();
     }
 
     @Test
     public void test_applyEither() throws Exception {
-        CompletableFuture nextStage = invocationFuture.applyToEither(newCompletedFuture(null),
-                (t) -> t);
-        invocationFuture.complete(response);
+        CompletableFuture nextStage = invocationFuture.applyToEither(newCompletedFuture(null), (t) -> t);
 
-        assertTrueEventually(() -> assertTrue(nextStage.isDone()));
         assertEquals(null, nextStage.get(10, TimeUnit.SECONDS));
         verify(callIdSequence).forceNext();
-        verify(callIdSequence, times(2)).complete();
+        verify(callIdSequence, times(1)).complete();
     }
 
     @Test
     public void test_runAfterBoth() throws Exception {
-        CompletableFuture nextStage = invocationFuture.runAfterBoth(newCompletedFuture(null), () -> ignore(null));
-        invocationFuture.complete(response);
+        CompletableFuture<Void> nextStage = invocationFuture.runAfterBoth(newCompletedFuture(null), () -> ignore(null));
+        invocationFuture.complete(null);
 
         assertEquals(null, nextStage.get(10, TimeUnit.SECONDS));
         verify(callIdSequence).forceNext();
@@ -309,12 +304,11 @@ public class ClientInvocationFutureTest {
 
     @Test
     public void test_runAfterEither() throws Exception {
-        CompletableFuture nextStage = invocationFuture.runAfterEither(newCompletedFuture(null),
+        CompletableFuture<Void> nextStage = invocationFuture.runAfterEither(newCompletedFuture(null),
                 () -> ignore(null));
-        invocationFuture.complete(response);
 
         assertEquals(null, nextStage.get(10, TimeUnit.SECONDS));
         verify(callIdSequence).forceNext();
-        verify(callIdSequence, times(2)).complete();
+        verify(callIdSequence, times(1)).complete();
     }
 }

